aya/src/maps/mod.rs

```diff
@@ -304,6 +317,45 @@ impl Map {
         Ok(fd)
     }
 
+    /// Loads a map from a pinned path in bpffs.
+    pub fn from_pinned<P: AsRef<Path>>(path: P) -> Result<Map, MapError> {
+        let path_string = CString::new(path.as_ref().to_str().unwrap()).unwrap();
```

Comment:
unwrapping in library code is only ok when 100% guaranteed not to panic. We should probably use `to_string_lossy()` here instead

Reply:
to_string_lossy() like so 

```rust
let path_string = CString::new(path.as_ref().to_string_lossy())
        .map_err(|e| MapError::ParsePinError{ error: PinError::InvalidPinPath { error: e.to_string() } })?;
```

Results in 

```rust 
 error[E0277]: the trait bound `Vec<u8>: From<Cow<'_, str>>` is not satisfied
   --> aya/src/maps/mod.rs:320:40
    |
320 |         let path_string = CString::new(path.as_ref().to_string_lossy())
    |                           ------------ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<Cow<'_, str>>` is not implemented for `Vec<u8>`
    |                           |
    |                           required by a bound introduced by this call
    |
    = help: the following other types implement trait `From<T>`:
```       

So Is this how you envisioned it being used?

Reply:
`to_string_lossy().into_owned()` and you can use the existing `MapError::PinError` (the name field probably needs to be changed to `Option<String>`)

Reply:
Yep cool!